iosTraining-Reeen/APICaller.swift

```diff
@@ -9,11 +9,23 @@ import Foundation
 import YumemiWeather
 
 protocol WeatherServiceProtocol {
-    func getWeatherInformation() -> String
+    var delegate: WeatherServiceDelegate? { get set }
+
+    func getWeatherInformation()
+}
+
+protocol WeatherServiceDelegate: AnyObject {
+    func weatherService(_ weatherService: WeatherServiceProtocol, conditonUpdate weatherInfo: String)
```

Comment:
[should]
「何が起きたか」を伝えるという意味ではもうちょっと改良できそうです。
あと typo してました。 conditon → condition

```suggestion
    func weatherService(_ weatherService: WeatherServiceProtocol, didUpdateCondition weatherInfo: String)
```

ここまで書いて思ったのですが、状況としては "condition が更新 (Update) された" というよりは "condition を取得した" かなと思ったので、 `didFetchCondition` も良さそうです。

Reply:
確かにUITableViewの`didSelectRowAt`と同様に、`conditonUpdate`よりも`didUpdateCondition`の方が見やすいなと思いました。

Reply:
こちらで修正しました。
https://github.com/yumemi-inc/iosTraining-Reeen/pull/4/commits/68b9c3e38f325010784f5a173d2a15a583a23fb2